src/java.base/share/classes/java/nio/file/spi/FileSystemProvider.java

```diff
@@ -1170,4 +1170,122 @@ public abstract Map<String,Object> readAttributes(Path path, String attributes,
     public abstract void setAttribute(Path path, String attribute,
                                       Object value, LinkOption... options)
         throws IOException;
+
+    /**
+     * Tests whether a file exists. This method works in exactly the
+     * manner specified by the {@link Files#exists(Path, LinkOption...)} method.
+     *
+     * @implSpec
+     * The default implementation of this method invokes the
+     * {@link #checkAccess(Path, AccessMode...)} method when symbolic links
+     * are followed. If the option {@link LinkOption#NOFOLLOW_LINKS NOFOLLOW_LINKS}
+     * is present then symbolic links are not followed and the method
+     *  {@link #readAttributes(Path, Class, LinkOption...)} is called
+     *  to determine whether a file exists.
+     *
+     * @param   path
+     *          the path to the file to test
+     * @param   options
+     *          options indicating how symbolic links are handled
+     *
+     * @return  {@code true} if the file exists; {@code false} if the file does
+     *          not exist or its existence cannot be determined.
+     *
+     * @throws  SecurityException
+     *          In the case of the default provider, the {@link
+     *          SecurityManager#checkRead(String)} is invoked to check
+     *          read access to the file.
+     *
+     * @since 20
+     */
+    public boolean exists(Path path, LinkOption... options) {
+        try {
+            if (followLinks(options)) {
+                this.checkAccess(path);
+            } else {
+                // attempt to read attributes without following links
+                readAttributes(path, BasicFileAttributes.class,  LinkOption.NOFOLLOW_LINKS);
+            }
+            // file exists
+            return true;
+        } catch (IOException x) {
+            // does not exist or unable to determine if file exists
+            return false;
+        }
+    }
+
+    /**
+     * Reads a file's attributes as a bulk operation if it exists.
```

Comment:
Maybe instead "it" put "the file"?

Reply:
I can if you thank that is better.  I followed the wording for `FileSystemProvider::deleteIfExists` trying to be consistent

Reply:
Then leave is as is.